### PR TITLE
add annotation

### PR DESCRIPTION
**Pull request by leoscholl**
utility to add annotations to spatial maps

### CODE REVIEW

File: aopy/visualization/base.py [hunk 1]
```diff
@@ -321,6 +321,7 @@ def plot_ECoG244_data_map(data, bad_elec=[], interp=True, cmap='bwr', ax=None, *
         bad_elec (list, optional): channels to remove from the plot. Defaults to [].
         interp (bool, optional): flag to include 2D interpolation of the result. Defaults to True.
         cmap (str, optional): matplotlib colormap to use in image. Defaults to 'bwr'.
+        theta (float, optional): rotation in degrees to apply to channel mapping. Default 0.
```
Comment (pavi-rajes) [inline aopy/visualization/base.py:324]: Some comments about the direction of rotation.  
```suggestion
        theta (float, optional): rotation in degrees to apply to channel mapping. Default 0. rotations are applied clockwise. Ex: , theta = 90 rotates the map clockwise by 90 degrees, -90 rotates the map anti-clockwise by 90 degrees.
```

File: aopy/visualization/base.py [hunk 2]
```diff
@@ -371,6 +372,65 @@ def plot_ECoG244_data_map(data, bad_elec=[], interp=True, cmap='bwr', ax=None, *
...
+        plt.Annotation: annotation object
+    '''
+    if ax is None:
+        ax = plt.gca()
+    return ax.annotate(text, elec_pos, color=color, fontsize=fontsize, ha='center', va='center', **kwargs)
+    
+def annotate_spatial_map_channels(acq_ch=None, drive_type='ECoG244', theta=0, color='k', fontsize=6, 
+                                  print_zero_index=True, ax=None, **kwargs):
+    '''
+    Prints 0-index channel by default
```
Comment (pavi-rajes) [inline aopy/visualization/base.py:397]:
```suggestion
    Prints 0-index channel by default. 
    Note: acq_ch from `func::aopy.data.load_chmap` is a 1-indexed list of acquisition channels connected to electrodes. Choose print_zero_index accordingly. 
```

File: aopy/visualization/base.py [hunk 3]
```diff
@@ -371,6 +372,65 @@ def plot_ECoG244_data_map(data, bad_elec=[], interp=True, cmap='bwr', ax=None, *
...
+            aopy.visualization.annotate_spatial_map_channels(acq_ch=None, drive_type='ECoG244', color='k')
+            aopy.visualization.annotate_spatial_map_channels(acq_ch=None, drive_type='Opto32', color='b', print_zero_index=False)
+            plt.axis('off')
+
+        .. image:: _/images/ecog244_opto32.png
+    '''
+    if ax is None:
+        ax = plt.gca()
+    if acq_ch is not None:
+        acq_ch = np.array(acq_ch)+1 # Change from 0 to 1 index
```
Comment (pavi-rajes) [inline aopy/visualization/base.py:425]: this is a bit confusing. Why change zero indexed channels (docstrings say it has to be zero indexed) to 1 indexed here?

File: aopy/visualization/base.py [hunk 4]
```diff
@@ -371,6 +372,65 @@ def plot_ECoG244_data_map(data, bad_elec=[], interp=True, cmap='bwr', ax=None, *
...
+    elec_pos, acq_ch, elecs = load_chmap(drive_type, acq_ch, theta)
```
Comment (pavi-rajes) [inline aopy/visualization/base.py:426]: Currently, this line overwrites previous values for acq_ch. Should it be in else?

**Review by pavi-rajes — COMMENTED** [review]
Overall looks good. Made some minor suggestions for docstrings. I am a bit confused about changing 0 to 1 index in acq-ch and overwriting it in the next sentence.

File: aopy/visualization/base.py (hunk 4 above)
Comment (pavi-rajes) [reply aopy/visualization/base.py:426]: Will approve this PR once this is clear. :)

**Comment by leoscholl** [comment]
i updated comments and docstrings a bit to make things more clear. i made the input `acq_ch` always 0-indexed for no particular reason. would it be simpler to also be 1-indexed like the channel mappings?

File: aopy/visualization/base.py [hunk 5]
```diff
@@ -418,12 +420,22 @@ def annotate_spatial_map_channels(acq_ch=None, drive_type='ECoG244', theta=0, co
...
+    Note: 
+        The acq_ch returned from `func::aopy.data.load_chmap` are 1-indexed lists of acquisition 
+        channels connected to electrodes. However, the acq_ch input to this function is always 
+        0-indexed (or None for all channels). The print_zero_index argument controls in what format the
+        channels are printed onto the figure.
     '''
     if ax is None:
         ax = plt.gca()
     if acq_ch is not None:
         acq_ch = np.array(acq_ch)+1 # Change from 0 to 1 index
```
Comment (pavi-rajes) [inline aopy/visualization/base.py:433]: change this to: 
```
if acq_ch is not None: 
       acq_ch = np.array(acq_idx) + 1
```

Comment (pavi-rajes) [reply aopy/visualization/base.py:433]: plan is to have two different input variables for acq_ch vs acq_idx. That would be less confusing and will prevent unintentional -1 or +1.

**Review by pavi-rajes — APPROVED** [review]
changes look good.

**Pull request merged**